IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m

```diff
@@ -367,7 +367,7 @@ - (void)decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
         return;
     }
     
-    if ([requestURLString isEqualToString:@"about:blank"])
+    if ([requestURLString isEqualToString:@"about:blank"] || [requestURLString isEqualToString:@"about:srcdoc"])
```

Comment:
Maybe we should expand it to all about: scheme patterns-: https://en.wikipedia.org/wiki/About_URI_scheme

Reply:
I see about:srcdoc is used for iframes. I'm not sure iframes are secure, we should confirm if they can be present in auth flow.

Reply:
The first question about allow all `about:` I see too many values in there, we shouldn't allow all.
About the 2nd question, Here is for more information: 
`https://chromium.googlesource.com/chromium/src.git/+/master/docs/special_case_urls.md`
Could you elaborate more about your concern if adding `about:srcdoc` beside `about:blank`? There is a provider (facebook) using `about:srcdoc`, so I was trying to unblock customer. But if it exposes security concerns, we can revise.

Reply:
about:srcdoc should be unresolvable. Can you clarify what will happen if we perform WKNavigationActionPolicyAllow for it? Would it be just a blank page?

Reading from [html standard](https://html.spec.whatwg.org/multipage/urls-and-fetching.html#terminology-2):

`This specification defines the URL about:srcdoc as a reserved, though unresolvable, about: URL, that is used as the URL of iframe srcdoc documents. [ABOUT]`

Reply:
There is a trick that declares the whole page inside an iFrame tag, and I think that's why we see that URL in our webview controller. The motivation is unclear to me why they want to do that. What happens on the website is totally controlled by the IdP (in this case Facebook), we have too little information to do anything with it, other than forcing https.

Reply:
Ok, I'm fine with the change in that case.

Reply:
According to [w3c](https://w3c.github.io/webappsec-secure-contexts/#is-url-trustworthy) about:srcdoc is potentially trustworthy on condition that top level frame is https and not http. Since our navigation policy allows only https, allowing about:srcdoc should be fine.